src/utils/featheryClient/index.ts

```diff
@@ -454,10 +454,10 @@ export default class FeatheryClient extends IntegrationClient {
    */
   async _debouncedSubmitCustom(override: boolean) {
     if (Object.keys(this.pendingCustomFieldUpdates).length === 0) {
-      // if no pending changes, no need to keep listening for unload events.
-      this._removeCustomFieldListener();
       return;
     }
+    // if there are changes, listen for beforeunload events
+    this._addCustomFieldListener();
```

Comment:
if debounced function continues after the pending changes check, we can add the before unload listener